release/nightly_tests/dataset/data_ingest_benchmark.py

```diff
@@ -104,28 +141,48 @@ def make_ds(size_gb: int, parallelism: int = -1):
     return dataset
 
 
-def run_ingest_streaming(dataset_size_gb, num_workers):
+def run_ingest_streaming(dataset_size_gb, num_workers, use_gpu, early_stop):
     ds = make_ds(dataset_size_gb)
+    resources = {"num_cpus": 0.5}
+    if use_gpu:
+        resources["num_gpus"] = 0.5
     consumers = [
-        ConsumingActor.options(scheduling_strategy="SPREAD").remote(i)
+        ConsumingActor.options(scheduling_strategy="SPREAD", **resources).remote(i)
         for i in range(num_workers)
     ]
     locality_hints = ray.get([actor.get_location.remote() for actor in consumers])
     ds = ds.map_batches(lambda df: df * 2, batch_format="pandas")
     splits = ds.streaming_split(num_workers, equal=True, locality_hints=locality_hints)
-    future = [consumers[i].consume.remote(s) for i, s in enumerate(splits)]
+    expected_total_read_bytes = dataset_size_gb * GiB
+    early_stop_when_read_bytes = None
+    if early_stop:
+        early_stop_when_read_bytes = expected_total_read_bytes // 2
+        expected_total_read_bytes //= 2
+    # Early stop when we've read half the dataset.
```

Comment:
~~Why're we doing early stopping? Also, any reason we're stopping at half in specific?~~

Ah, nevermind. Saw your comment in `release_tests.yaml`